src/plugins/data/public/index_patterns/saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { omit } from 'lodash';
+import { SavedObjectsClient, SimpleSavedObject } from 'src/core/public';
+import {
+  SavedObjectsClientCommon,
+  SavedObjectsClientCommonFindArgs,
+  SavedObject,
+} from '../../common/index_patterns';
+
+type SOClient = Pick<SavedObjectsClient, 'find' | 'get' | 'update' | 'create' | 'delete'>;
+
+const simpleSavedObjectToSavedObject = <T>(
+  simpleSavedObject: SimpleSavedObject
+): SavedObject<T> => ({
+  version: simpleSavedObject._version,
+  ...omit(simpleSavedObject, '_version'),
+});
+
+export class SavedObjectsClientPublicToCommon implements SavedObjectsClientCommon {
```

Comment:
have you checked with platform team about this ?

Reply:
Yes, they've confirmed that its the correct strategy and that more complete parity between APIs would take a fair amount more work.